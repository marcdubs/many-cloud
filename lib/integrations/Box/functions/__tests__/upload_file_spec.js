describe("upload_file", () => {
  const expected_resolution = "Some file info";
  let data, func;

  beforeEach(() => {
    data = {
      client: {
        files: {
          uploadFile: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              resolve(expected_resolution);
            });
          })
        }
      }
    };
    func = require("../upload_file")(data);
  });

  describe("when provided a file", () => {
    let actual;
    beforeEach(async () => {
      actual = await func(0, "dummy_files/TestFile.txt");
    });

    it("calls data.client.uploadFile", () => {
      expect(data.client.files.uploadFile).toHaveBeenCalled();
    });

    it("returns the expected file result", () => {
      expect(actual).toEqual(expected_resolution);
    });
  });

  describe("when provided a folder", () => {
    it("alerts the user to use upload_folder instead", async () => {
      await expect(
        func(0, "dummy_files")
      ).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe("when not provided a file path", () => {
    it("rejects with an error", async () => {
      await expect(func(0, null)).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe("when uploading the file returns an error", () => {
    beforeEach(() => {
      data = {
        client: {
          files: {
            uploadFile: jest.fn().mockImplementation(() => {
              return new Promise((resolve, reject) => {
                throw new Error(
                  "Something absolutely terrible happened. Oh no."
                );
              });
            })
          }
        }
      };
      func = require("../upload_file")(data);
    });

    it("rejects with the error", async () => {
      await expect(
        func(0, "dummy_files/TestFile.txt")
      ).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
