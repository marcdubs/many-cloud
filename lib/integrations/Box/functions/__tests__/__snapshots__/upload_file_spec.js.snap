// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upload_file when not provided a file path rejects with an error 1`] = `"Missing file path parameter."`;

exports[`upload_file when provided a folder alerts the user to use upload_folder instead 1`] = `"Path represents a directory. Please use upload_folder."`;

exports[`upload_file when uploading the file returns an error rejects with the error 1`] = `"Something absolutely terrible happened. Oh no."`;
